usr/bin/regolith-session-init

```diff
@@ -5,28 +5,47 @@ set -Eeu -o pipefail
 
 source /usr/lib/regolith/regolith-session-common.sh
 
-# --------- Script Main -----------
-
-load_standard_xres  # ~/.Xresources
-load_regolith_xres  # ~/.config/regolith2/Xresources
-
-# Register with gnome-session so that it does not kill the whole session thinking it is dead.
-# See https://zork.net/~st/jottings/gnome-i3.html for details
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.RegisterClient "string:Regolith" "string:$DESKTOP_AUTOSTART_ID"
-fi
+start_i3 () {
+    load_standard_xres  # ~/.Xresources
+    load_regolith_xres  # ~/.config/regolith2/Xresources
+
+    # Register with gnome-session so that it does not kill the whole session thinking it is dead.
+    # See https://zork.net/~st/jottings/gnome-i3.html for details
+    if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
+        dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.RegisterClient "string:Regolith" "string:$DESKTOP_AUTOSTART_ID"
+    fi
+
+    echo "Regolith is launching i3 with $I3_CONFIG_FILE"
+    i3 -c "$I3_CONFIG_FILE"
+
+    # Run user's post logout script if script. Post logout means that the X session is closed at this point.
+    if [ -f "$USER_POST_LOGOUT_SCRIPT_FILE" ] ; then
+        echo "Regolith is launching user post-logout file $USER_POST_LOGOUT_SCRIPT_FILE"
+        timeout --verbose 5 bash "$USER_POST_LOGOUT_SCRIPT_FILE"
+    fi
+
+    # Close session when i3 exits.
+    if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
+        dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.Logout "uint32:1"
+    fi
+}
+
+start_sway () {
+    load_standard_trawlres
+    load_regolith_trawlres
+    
+    echo "Regolith is launching sway with $SWAY_CONFIG_FILE"
+    sway -c "$SWAY_CONFIG_FILE"
+}
 
+# --------- Script Main -----------
 resolve_default_config_file
-echo "Regolith is launching i3 with $I3_CONFIG_FILE"
-i3 -c "$I3_CONFIG_FILE"
-
-# Run user's post logout script if script. Post logout means that the X session is closed at this point.
-if [ -f "$USER_POST_LOGOUT_SCRIPT_FILE" ] ; then
-    echo "Regolith is launching user post-logout file $USER_POST_LOGOUT_SCRIPT_FILE"
-	timeout --verbose 5 bash "$USER_POST_LOGOUT_SCRIPT_FILE"
-fi
-
-# Close session when i3 exits.
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.Logout "uint32:1"
+export MOZ_ENABLE_WAYLAND=1
+export STUDIO_JDK=/usr/lib/jvm/java-11-openjdk/
```

Comment:
Hmm..what is this?

Reply:
Some applications require additional environment variables to work well with Wayland. It's quite common to have such variable definitions included in starter scripts. And these variables need to be declared before sway starts.

Reply:
The second line is necessary for a lot of Java based applications (especially for Intellij based IDEs) to work properly.

Reply:
Although, I think it might be a good idea to let users define their pre-startup environment variables at one place. What are you're thoughts on this?

Reply:
My opinion is app specific logic should not live in general-purpose scripts.  For a user that does not use Firefox (assuming that from `Moz`) or care about java programs would classify this as bloat IMO.  Also, what's to say that the JVM path is the one a user really wants Inteliij to point to, or is even valid?

The simplest thing we can do is provide guidance/docs for poorly behaving apps that users encounter.  More work would be to create packages that install support for some applications like these.  I'm guessing there is something in Wayland like `/etc/X11/Xsession` that allows the user environment to be extended by a package.